src/measure_theory/function/conditional_expectation/basic.lean

```diff
@@ -2080,6 +2080,48 @@ begin
   rw [hg_eq s hs hμs, set_integral_condexp hm hf hs],
 end
 
+lemma condexp_bot' [hμ : μ.ae.ne_bot] (f : α → F') :
+  μ[f|⊥] = λ _, (μ set.univ).to_real⁻¹ • ∫ x, f x ∂μ :=
+begin
+  by_cases hμ_finite : is_finite_measure μ,
+  swap,
+  { have h : ¬ sigma_finite (μ.trim bot_le),
+    { rwa sigma_finite_trim_bot_iff, },
+    rw not_is_finite_measure_iff at hμ_finite,
+    rw [condexp_of_not_sigma_finite bot_le h],
+    simp only [hμ_finite, ennreal.top_to_real, inv_zero, zero_smul],
+    refl, },
+  haveI : is_finite_measure μ := hμ_finite,
+  by_cases hf : integrable f μ,
+  swap, { rw [integral_undef hf, smul_zero, condexp_undef hf], refl, },
+  have h_meas : strongly_measurable[⊥] (μ[f|⊥]) := strongly_measurable_condexp,
+  obtain ⟨c, h_eq⟩ := strongly_measurable_bot_iff.mp h_meas,
+  rw h_eq,
+  have h_integral : ∫ x, μ[f|⊥] x ∂μ = ∫ x, f x ∂μ := integral_condexp hf,
+  swap, { exact bot_le, },
+  swap, { rw sigma_finite_trim_bot_iff, exact hμ_finite, },
```

Comment:
The fact that you have these side goals here and that you need these `swap` is because, in `integral_condexp`, the argument `hm : m ≤ m0` is implicit for no good reason. Could you make it explicit?

Reply:
Done!